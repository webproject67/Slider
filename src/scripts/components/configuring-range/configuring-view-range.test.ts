import SliderModel from '../model/slider-model';
import ConfiguringViewRange from './configuring-view-range';

const main = document.createElement('div');
main.id = 'slider'
const state = {
  flag: true,
  from: -10000,
  fromPercent: 0,
  max: 100,
  min: 0,
  range: 'range',
  scale: true,
  step: 1,
  to: -10000,
  toPercent: 100,
  view: 'horizontal'
};

const model = new SliderModel(main, state);
const configuring = new ConfiguringViewRange(model);

test('spyOn change input', () => {
  const somethingSpy = jest.spyOn(configuring, 'onInputChange');
  let evt: any;
  configuring.onInputChange(evt);
  expect(somethingSpy).toHaveBeenCalledTimes(1)
});

it('jest snapshots', () => {
  expect(configuring.element).toMatchSnapshot()
})