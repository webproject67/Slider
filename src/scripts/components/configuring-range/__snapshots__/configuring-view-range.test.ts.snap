// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest snapshots 1`] = `
<div
  class="slider__inputs"
>
  
      
  <div
    class="slider__input"
    data-testid="configuringRange"
  >
    
        
    <label
      class="slider__label"
      for="min"
    >
      Минимальное значение
    </label>
    
        
    <input
      class="slider__min slider__number"
      data-name="min"
      id="min"
      type="number"
      value="0"
    />
    
      
  </div>
  
      
  <div
    class="slider__input"
  >
    
        
    <label
      class="slider__label"
      for="max"
    >
      Максимальное значение
    </label>
    
        
    <input
      class="slider__max slider__number"
      data-name="max"
      id="max"
      type="number"
      value="100"
    />
    
      
  </div>
  
      
  <div
    class="slider__input"
  >
    
        
    <label
      class="slider__label"
      for="from"
    >
      От
    </label>
    
        
    <input
      class="slider__from slider__number"
      data-name="from"
      id="from"
      readonly=""
      type="number"
      value="0"
    />
    
      
  </div>
  
      
  <div
    class="slider__input"
  >
    
        
    <label
      class="slider__label"
      for="to"
    >
      До
    </label>
    
        
    <input
      class="slider__to slider__number"
      data-name="to"
      id="to"
      readonly=""
      type="number"
      value="100"
    />
    
      
  </div>
  
      
  <div
    class="slider__input"
  >
    
        
    <label
      class="slider__label"
      for="step"
    >
      Шаг
    </label>
    
        
    <input
      class="slider__step slider__number"
      data-name="step"
      id="step"
      type="number"
      value="1"
    />
    
      
  </div>
  
      
  <div
    class="slider__input"
  >
    
        
    <input
      checked=""
      class="slider__view"
      data-name="view"
      id="horizontal#slider"
      name="view#slider"
      type="radio"
      value="horizontal"
    />
    
        
    <label
      for="horizontal#slider"
    >
      Горизонтальный
    </label>
    
        
    <input
      class="slider__view"
      data-name="view"
      id="vertical#slider"
      name="view#slider"
      type="radio"
      value="vertical"
    />
    
        
    <label
      for="vertical#slider"
    >
      Вертикальный
    </label>
    
      
  </div>
  
      
  <div
    class="slider__input"
  >
    
        
    <input
      checked=""
      class="slider__range"
      data-name="range"
      id="one#slider"
      name="range#slider"
      type="radio"
      value="one"
    />
    
        
    <label
      for="one#slider"
    >
      Одиночное значение
    </label>
    
        
    <input
      class="slider__range"
      data-name="range"
      id="range#slider"
      name="range#slider"
      type="radio"
      value="range"
    />
    
        
    <label
      for="range#slider"
    >
      Интервал
    </label>
    
      
  </div>
  
      
  <div
    class="slider__input"
  >
    
        
    <input
      checked=""
      class="slider__flag-checkbox"
      data-name="flag"
      id="flag#slider"
      type="checkbox"
    />
    
        
    <label
      for="flag#slider"
    >
      Значение
    </label>
    
      
  </div>
  
      
  <div
    class="slider__input"
  >
    
        
    <input
      checked=""
      class="slider__scale-checkbox"
      data-name="scale"
      id="scale#slider"
      type="checkbox"
    />
    
        
    <label
      for="scale#slider"
    >
      Шкала
    </label>
    
      
  </div>
  
    
</div>
`;
